src/cdomains/intDomain.ml

```diff
@@ -642,11 +645,24 @@ struct
   let cast_to ?torg ?no_ov t = norm ~cast:true t (* norm does all overflow handling *)
 
   let widen ik x y =
+    let threshold = get_bool "ana.int.interval_threshold_widening" in
+    let upper_threshold u =
+      let ts = Lazy.force widening_thresholds in
+      let t = List.find_opt (fun x -> Ints_t.compare u (Ints_t.of_bigint x) <= 0) ts in
```

Comment:
How would this be done short of mapping (potentially needlessly) the entire list to `Ints_t.t`?

Reply:
Ahh, sorry, I think I got confused about which variable was bound by the lambda.
I suppose the other possibility would be to convert `u` to bigint once and then do all the comparisons with `BI.compare`.
Although I guess most of the time this doesn't even matter since `Ints_t` ends up being `BI` anyway.

Reply:
I now went with your suggestion.

Reply:
This way the `Ints_t.to_bigint u` can be extracted out, like I initially had imagined :D

Reply:
@michael-schwarz would you mind doing this?

Reply:
done!